### ISSUE TITLE
Is there a way to dynamically add a signal to mix?

### ISSUE THREAD

**Issue by TonalidadeHidrica** [RustAudio/dasp#157]
I want to play a music and some sound effects.  Can I dynamically add another `Signal` to mix while playing?  As I see the `Signal` API, it seems to be designed for static mixing.

**Comment by mitchmindtree** [comment]
@TonalidadeHidrica that's right, using the `Signal` API it might be easier to reconstruct the full signal when you want to add or remove `Signal`s dynamically. Alternatively, you could implement `Signal` for your own "signal collection" type that allows for adding/removing signals.

For dynamic composition of audio graphs, you might want to consider the [dasp_graph](https://docs.rs/dasp_graph/0.11.0/dasp_graph/) crate - you can find a section in the docs discussing the difference to the dasp `Signal` API [here](https://docs.rs/dasp_graph/0.11.0/dasp_graph/#comparison-to-dasp_signal).
